src/Compilers/CSharp/Portable/FlowAnalysis/VariablesDeclaredWalker.cs

```diff
@@ -131,13 +131,7 @@ public override void VisitForEachIterationVariable(BoundForEachStatement node)
                     _variablesDeclared.Add(node.IterationVariableOpt);
                 }
 
-                var leftVariables =
-                    (node.DeconstructionOpt?.DeconstructionAssignment?.LeftVariables)
-                    .GetValueOrDefault().NullToEmpty();
-                foreach (var variable in leftVariables)
-                {
-                    Visit(variable);
-                }
+                node.DeconstructionOpt?.DeconstructionAssignment?.Left?.VisitAllElements((x, self) => self.Visit(x), this);
```

Comment:
`?`  is only needed after optional items. Things that cannot return null, should not be null-checked.